lib/_http_client.js

```diff
@@ -24,6 +24,18 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
+  } else if (options instanceof url.URL) {
+    options = {
+      protocol: options.protocol,
+      host: options.host,
+      hostname: options.hostname,
+      port: options.port,
+      hash: options.hash,
+      search: options.search,
+      pathname: options.pathname,
+      path: `${options.pathname}${options.search}`,
+      href: options.href
+    };
```

Comment:
Also – could this blocked be just executed independently of `options instanceof url.URL`? It would mean using `path: options.path || …`, but other than that, it should just work, no?

(If you feel unsure about these ideas, I can PR them myself later. If you feel they are horrible, say so and I won’t. ;))